crates/storage/libmdbx-rs/src/environment.rs

```diff
@@ -765,8 +859,123 @@ impl EnvironmentBuilder {
         self
     }
 
+    /// Set the Handle-Slow-Readers callback. See [HandleSlowReadersCallback] for more information.
+    pub fn set_handle_slow_readers(&mut self, hsr: HandleSlowReadersCallback) -> &mut Self {
+        self.handle_slow_readers = Some(hsr);
+        self
+    }
+
     pub fn set_log_level(&mut self, log_level: ffi::MDBX_log_level_t) -> &mut Self {
         self.log_level = Some(log_level);
         self
     }
 }
+
```

Comment:
this should mention that this leaks memory, but it's fine because it's expected that environment is alive for the duration of the entire program

Reply:
ok i think the proper way would be to make the closure live as long as the environment is alive, without leaking for the whole duration of the program

Reply:
ah, it will require adding a lifetime on the `Environment` struct... Added a comment about leaking.